packages/contractkit/src/utils/sign-typed-data-utils.ts

```diff
@@ -0,0 +1,112 @@
+import { BigNumber } from 'bignumber.js'
+import * as ethUtil from 'ethereumjs-util'
+import coder from 'web3-eth-abi'
+
+export interface EIP712Parameter {
+  name: string
+  type: string
+}
+
+export interface EIP712Types {
+  [key: string]: EIP712Parameter[]
+}
+
+export type EIP712ObjectValue = string | number | EIP712Object
+
+export interface EIP712Object {
+  [key: string]: EIP712ObjectValue
+}
+
+export interface EIP712TypedData {
+  types: EIP712Types
+  domain: EIP712Object
+  message: EIP712Object
+  primaryType: string
+}
+
+/**
+ * Generates the EIP712 Typed Data hash for signing
+ * @param   typedData An object that conforms to the EIP712TypedData interface
+ * @return  A Buffer containing the hash of the typed data.
+ */
+export function generateTypedDataHash(typedData: EIP712TypedData): Buffer {
+  return ethUtil.sha3(
+    Buffer.concat([
+      Buffer.from('1901', 'hex'),
+      _structHash('EIP712Domain', typedData.domain, typedData.types),
+      _structHash(typedData.primaryType, typedData.message, typedData.types),
+    ])
+  ) as Buffer
+}
+
+function _findDependencies(
```

Comment:
Nit: no need for leading underscores. If the functions aren't exported then they're private to this module anyway

Reply:
Yes! My bad